packages/compiler/src/lib/decorators.ts

```diff
@@ -164,17 +165,30 @@ export function getIndexer(program: Program, target: Type): ModelIndexer | undef
 export function isStringType(program: Program | ProjectedProgram, target: Type): target is Scalar {
   const coreType = program.checker.getStdType("string");
   const stringType = target.projector ? target.projector.projectType(coreType) : coreType;
-  return (
-    target.kind === "Scalar" && program.checker.isTypeAssignableTo(target, stringType, target)[0]
-  );
+  return isOrUnionOfOrNull(target, (type) => {
+    return type.kind === "Scalar" && program.checker.isTypeAssignableTo(type, stringType, type)[0];
+  });
 }
 
-export function isNumericType(program: Program | ProjectedProgram, target: Type): target is Scalar {
+export function isNumericType(program: Program | ProjectedProgram, target: Type): boolean {
   const coreType = program.checker.getStdType("numeric");
   const numericType = target.projector ? target.projector.projectType(coreType) : coreType;
-  return (
-    target.kind === "Scalar" && program.checker.isTypeAssignableTo(target, numericType, target)[0]
-  );
+  return isOrUnionOfOrNull(target, (type) => {
+    return type.kind === "Scalar" && program.checker.isTypeAssignableTo(type, numericType, type)[0];
+  });
+}
+
+/**
+ * Check the given type is matching the given condition or is a union of null and types matching the condition.
+ * @param type Type to test
+ * @param condition Condition
+ * @returns Boolean
+ */
+function isOrUnionOfOrNull(type: Type, condition: (type: Type) => boolean): boolean {
```

Comment:
This helper is so confusingly named 😂 I wonder, do we even need it? I feel like this amounts to an assignability check to a union of the condition types. e.g. if the max length decorator target is assignable to `string | null`, you're good, which handles the single `string` case as well as larger unions of string subtypes, etc. It would also allow `maxLength` on a type like `string | number`, which I can't recall if I filed, but would also be nice (and is something OpenAPI supports).

Reply:
I think this is fine, as long as the constraint applies to at least one variant, personally.